echo-rest/src/test/groovy/com/netflix/spinnaker/echo/events/RestEventListenerSpec.groovy

```diff
@@ -34,18 +34,40 @@ class RestEventListenerSpec extends Specification {
     listener.eventName = 'defaultEvent'
     listener.fieldName = 'defaultField'
     listener.restUrls = new RestUrls()
+    listener.restEventTemplateEngine = new SimpleEventTemplateEngine()
     restService = Mock(RestService)
   }
 
+  void 'render template when template is set'() {
+    given:
+    listener.restUrls.services = [
+        [
+            client: restService,
+            config: [
+                template: '{"myCustomEventField":${event}}',
+                wrap    : true
+            ]
+        ]
+    ]
+
+    when:
+    listener.processEvent(event)
+
+    then:
+    1 * restService.recordEvent({
+      it.myCustomEventField == listener.mapper.convertValue(event, Map)
```

Comment:
Would rather see the assertion compare the value of `myCustomEventField` with a literal map, instead of relying on `listener.mapper.convertValue(event, Map)`.